trin-execution/src/execution.rs

```diff
@@ -104,7 +104,17 @@ impl TrinExecution {
                 if stop_signal_received {
                     info!("Stop signal received. Committing now, please wait!");
                 }
-                return self.commit(&block.header, block_executor).await;
+
+                let result = self.commit(&block.header, block_executor).await?;
+
+                if stop_signal_received {
```

Comment:
I think you can remove this check and log your message all the time (meaning, even if we reached the last_block).